src/v/datalake/translation/deps.h

```diff
@@ -109,6 +109,11 @@ class data_source {
      * Returns the next offset for translation, either
      * min_offset_for_translation, next(last_translated_offset) if one exists,
      * or nullopt if there is no data to translate.
+     * TODO:
+     * if we decide not to return on any data appearing we need to add a
+     * callback to the method ss::noncopyable_function<void(kafka::offset)>
+     * that would notify the lag tracker about new offsets that are ready to be
+     * translated.
```

Comment:
q: is this in reference to the "wait for enough data to appear" todo?

Reply:
exactly